x/incentives/keeper/gauge_test.go

```diff
@@ -582,3 +582,143 @@ func (s *KeeperTestSuite) TestCreateGauge_NoLockGauges() {
 		})
 	}
 }
+
+func (s *KeeperTestSuite) TestCreateGroupGauge() {
+	coinsToAdd := sdk.NewCoins(sdk.NewCoin("uosmo", sdk.NewInt(100_000_000)))
+	tests := []struct {
+		name             string
+		coins            sdk.Coins
+		numEpochPaidOver uint64
+		internalGaugeIds []uint64
+		expectErr        bool
+	}{
+		{
+			name:             "Happy case: created valid gauge",
+			coins:            coinsToAdd,
+			numEpochPaidOver: 1,
+			internalGaugeIds: []uint64{2, 3, 4},
+			expectErr:        false,
+		},
+
+		{
+			name:             "Error: Invalid InternalGauge Id",
+			coins:            coinsToAdd,
+			numEpochPaidOver: 1,
+			internalGaugeIds: []uint64{2, 3, 4, 5},
+			expectErr:        true,
+		},
+		{
+			name:             "Error: owner doesnot have enough funds",
+			coins:            sdk.NewCoins(sdk.NewCoin("uosmo", sdk.NewInt(200_000_000))),
+			numEpochPaidOver: 1,
+			internalGaugeIds: []uint64{2, 3, 4},
+			expectErr:        true,
+		},
```

Comment:
We should also probably cover the following two error cases:
1. One of the internal gauges is non-perpetual (internal gauges must be perpetual)
2. No internal gauges are specified

Reply:
added: https://github.com/osmosis-labs/osmosis/pull/6085/files#diff-3ed7186688da301d71a4300dfabcbfb27c2da4235d90c15100d3d5aad20e6a9bR618-R626